### DISCUSSION TITLE
Blank page after move to Phalcon 4 php 7.4

### DISCUSSION THREAD

**Discussion by alfeltrin** [phalcon/cphalcon#16002] [Q&A]
I have move an old application from a server with phalcon 3 and php 5 to another with phalcon 4 and php 7.4.
After that i get a blank screen and i don't know how to display the Phalcon errors.
I've check if phalcon is working and all wark fine with a simple 'Hello world'
But when i try to use Models, something goes wrong and i got a blank page instead of an error...there's a way to display the error message?

**Comment by quasipickle** [comment]
By default PHP is configured to hide errors.  Moving from 3 to 4, you almost certainly have fatal errors in your application - they're just not getting displayed.

Proper production environments should never display PHP errors.  However, if this is just internal or a development server showing the errors on the screen should be fine.

Find the `php.ini` file that is being used (`phpinfo()` on a page - not command line - will tell you that), then look for the `display_errors` directive and set it to "on".  Then, restart Apache & you should see a bunch of stuff on the screen telling you what to fix.

You should also read the upgrade guide: https://docs.phalcon.io/4.0/en/upgrade.

**Reply by alfeltrin** [reply]
I have enabled the display of errors inside index.php
`
error_reporting("E_ALL");
ini_set('display_errors', 1);
ini_set('display_startup_errors', 1);
`
However only some errors are displayed (it shows if it can't find a controller or an action inside the controller), but if the error occurre inside the controller code it shows a blank page

**Reply by quasipickle** [reply]
Changing the ini settings in-code requires the code to be fully parseable.  If there's a fatal error that happens during parsing or before your `ini_set()` calls get run, you won't see those errors

Changing the setting in `php.ini` will cause error reporting to be enabled _before_ your code gets parsed, so those errors will be caught & displayed.

**Comment by lieshanhe** [comment]
there are a variety of reasons cause the blank page.
1. Please check your web server log.
2. check Phalcon 4 response :

`if (!$response->isSent()) {
             $response->send();
        }
`
